packages/twenty-front/src/modules/workspace/hooks/useSubscriptionStatus.ts

```diff
@@ -0,0 +1,25 @@
+import { useRecoilValue } from 'recoil';
+
+import { useIsLogged } from '@/auth/hooks/useIsLogged';
+import { currentWorkspaceState } from '@/auth/states/currentWorkspaceState';
+import { billingState } from '@/client-config/states/billingState';
+import { SubscriptionStatus } from '~/generated/graphql';
+
+export const useSubscriptionStatus = ():
+  | SubscriptionStatus
+  | null
+  | undefined => {
+  const currentWorkspace = useRecoilValue(currentWorkspaceState);
+  const billing = useRecoilValue(billingState);
+  const isLoggedIn = useIsLogged();
+  if (!isLoggedIn) {
+    return undefined;
+  }
+  if (!billing?.isBillingEnabled) {
```

Comment:
i would not add this logic here that "lies" on the subscriptionStatus

In redirect conditions, we should check if billingIsEnabled && subscriptionStatus...